### PR TITLE
Fix SSH command blocking

### PR DESCRIPTION
**Pull request by coffeeavax**
https://github.com/ava-labs/avalanche-ops/issues/384

### CODE REVIEW

File: avalancheup-aws/src/apply/mod.rs [hunk 1]
```diff
@@ -1440,23 +1440,31 @@ pub async fn execute(log_level: &str, spec_file_path: &str, skip_prompt: bool) -
...
+                        Err(e) => log::warn!("failed to run ssh command {}", e),
                     }
-                    Err(e) => log::warn!("failed to run ssh command {}", e),
                 }
             }
+            let timeout_duration = Duration::from_secs(10);
+            let result = tokio::time::timeout(timeout_duration, read_script(ssh_commands)).await;
+            if result.is_err() {
+                log::warn!(
+                    "VPC security group has wrong IP for SSH2 22 port ingress traffic, please fix"
```
Comment (gyuho) [inline avalancheup-aws/src/apply/mod.rs:1465]:
```suggestion
                    "VPC security group may have wrong inbound rule for SSH 22 port, please fix"
```

File: avalancheup-aws/src/apply/mod.rs [hunk 2]
```diff
@@ -1440,23 +1440,31 @@ pub async fn execute(log_level: &str, spec_file_path: &str, skip_prompt: bool) -
...
+                            );
+                        }
+                        Err(e) => log::warn!("failed to run ssh command {}", e),
                     }
-                    Err(e) => log::warn!("failed to run ssh command {}", e),
                 }
             }
+            let timeout_duration = Duration::from_secs(10);
+            let result = tokio::time::timeout(timeout_duration, read_script(ssh_commands)).await;
+            if result.is_err() {
```
Comment (exdx) [inline avalancheup-aws/src/apply/mod.rs:1463]: This is ok but really Go style versus Rust. I would recommend using `?` or `.map_err()` for handling the error case.

File: avalancheup-aws/src/apply/mod.rs [hunk 3]
```diff
@@ -1440,23 +1440,31 @@ pub async fn execute(log_level: &str, spec_file_path: &str, skip_prompt: bool) -
...
-                        println!(
-                            "{} (anchor node) init script std output:\n{}\n",
-                            ssh_command.instance_id, output.stdout
-                        );
-                        println!(
-                            "{} (anchor node) init script std err:\n{}\n",
-                            ssh_command.instance_id, output.stderr
-                        );
+            sleep(Duration::from_secs(15)).await;          
+            async fn read_script(ssh_commands: Vec<ec2::SshCommand>) {
```
Comment (exdx) [inline avalancheup-aws/src/apply/mod.rs:1444]: Maybe we could make this a standalone function, for readability?

File: avalancheup-aws/src/apply/mod.rs (hunk 1 above)
Comment (exdx) [inline avalancheup-aws/src/apply/mod.rs:1465]: Can we make this error more descriptive? Like embedding the actual error, providing the wrong inbound rule as a string, etc.

**Review by exdx — COMMENTED** [review]
LGTM just a few minor comments.

File: avalancheup-aws/src/apply/mod.rs (hunk 2 above)
Comment (gyuho) [reply avalancheup-aws/src/apply/mod.rs:1463]: `?` o `map_err` will return here? The point here is to ignore the error

Comment (exdx) [reply avalancheup-aws/src/apply/mod.rs:1463]: hmm what about `.map_err(|_| log::warn!(...)`

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=43a8ac893765dca591ea8e717bf59831

**Review by gyuho — APPROVED** [review]

**Pull request merged**